From 06f2a05929e6afba9fe11ee5d006ea215dc80a6d Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 10 May 2022 09:49:08 +0300
Subject: [PATCH] [SW-66329] gaudi3: fix dual die sku logic

According to the gaudi3 efuse spec v0.5 the SKU bit indicates the
following:
0 - dual die SKU
1 - single die SKU

Adjusting the code logic accordingly.

Change-Id: I16cbf43bfbd4b94384e851e7a53563e930a64029
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 0bc492b..99907dd 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -243,7 +243,7 @@
 
 	val = (val >> EFUSE_SKU_SHIFT) & 0x1;
 
-	return !!val;
+	return !val;
 }
 
 #define GAUDI3_EFUSE_READ_RANGE_ALL	0xff00
